### PR TITLE
Add validation for draft annotations

### PR DESCRIPTION
**Pull request by asparuhft**
# Description

## What

- Add validation for draft annotations
- Replace logrus with ft logger
- Replace vestigo with mux

## Why

JIRA Ticket [UPPSF-4927](https://financialtimes.atlassian.net/browse/UPPSF-4927)

## Anything, in particular, you'd like to highlight to reviewers

Mention here sections of code which you would like reviewers to pay extra attention to .E.g

_Would appreciate a second pair of eyes on the test_  
_I am not quite sure how this bit works_  
_Is there a better library for doing x_  

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

## DoD - Ensure all relevant tasks are completed before marking this PR as "Ready for review"

- [ ] Test coverage is not significantly decreased
- [ ] All PR checks have passed
- [ ] Changes are deployed on dev before asking for review
- [ ] Documentations remains up-to-date
  - [ ] OpenAPI definition file is updated
  - [ ] README file is updated
  - [ ] Documentation is updated in upp-docs and upp-public-docs
  - [ ] Architecture diagrams are updated

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

[UPPSF-4927]: https://financialtimes.atlassian.net/browse/UPPSF-4927?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ManoelMilchev — APPROVED** [review]
LGTM!

File: handler/handler_test.go [hunk 1]
```diff
@@ -2176,29 +2382,31 @@ func TestHappyReplaceAnnotationWithPredicate(t *testing.T) {
...
 
-	h := handler.New(rw, annAPI, canonicalizer, aug, time.Second)
-	r := vestigo.NewRouter()
+	log := logger.NewUPPLogger("draft-annotations-api", "INFO")
+	v := validator.NewSchemaValidator(log).GetJSONValidator()
+	h := handler.New(rw, annAPI, canonicalizer, aug, v, time.Second, log)
+	r := mux.NewRouter()
 
-	r.Patch("/drafts/content/:uuid/annotations/:cuuid", h.ReplaceAnnotation)
+	r.HandleFunc("/drafts/content/{uuid}/annotations/{cuuid}", h.ReplaceAnnotation).Methods("PATCH")
```
[the remark this answers is not in the document]
Comment (asparuhft) [reply handler/handler_test.go]: Done

File: annotations/annotations_api.go [hunk 2]
```diff
@@ -68,13 +69,14 @@ type UPPAnnotationsAPI struct {
 	endpointTemplate string
 	username         string
 	password         string
-	httpClient       *http.Client
+	httpClient       *http.Client ``
```
[the remark this answers is not in the document]
Comment (asparuhft) [reply annotations/annotations_api.go]: It is some kind of leftover :)

**Review by ManoelMilchev — APPROVED** [review]
LGTM!

**Review by angelraynovft — APPROVED** [review]
LGTM

**Pull request merged**